IMC_R/EDA_Raymond/EDA-R.md

```diff
@@ -0,0 +1,123 @@
+
+Here, we started with our data frame, movies. Using the `%>%` operator introduced earlier, we can then group the data by genre with `group_by()`, then count the total number of observations in each group with `tally()`, and finally rearrange the results with `arrange(-n)`, where the `-n` specifies that we want the count column, n, to be in descending order.
+
+If we tweak this code a bit, we can investigate the data in more detail as well. Say, for example, you’re considering subscribing to Netflix. You’d be much more interested in what genres of movies are most common on Netflix, instead of in the entire dataset. We can simply add a filter to only include observations where the Netflix column has a value of 1.
+
+![](https://i.imgur.com/4j3Z5lC.png)
+
+Now if you were considering subscribing to any other streaming service, and you wanted to compare your options, you could run this with all the other sites as well to see which sites have the most movies from genres you enjoy. It would also be great to compare the size of their overall selections. We can get these numbers by filtering and running `summarise(count = n())`.
+
+![](https://i.imgur.com/kx96FBW.png)
+
+As we’ve seen, graphs and tables are great ways to understand specific characteristics of the data. And we can get quick answers to a lot of initial questions with Dplyr. But when we look at data, we are often interested in the relationships between variables as well. To get an example of this, let’s look at whether IMDb scores are correlated with Rotten Tomatoes scores.
+
+## IMDb vs Rotten Tomatoes
+IMDb and Rotten Tomatoes are two of the most popular sites for movie ratings. Given that they both perform a similar function, it would be reasonable to expect that the score a movie receives on IMDb would be correlated to the score it receives on Rotten Tomatoes, despite the fact that their scoring systems are quite different (IMDb scores come from audiences, while Rotten Tomatoes scores come from critics).
+
+Before we begin, let’s take a look at the distribution of scores from each site.
+
+![](https://i.imgur.com/u7A7KRv.png)![](https://i.imgur.com/QgiMv0U.png)
+
+Right away, something seems very wrong. The histogram of IMDb scores looks great, but there seems to be far too many near 0 observations for Rotten Tomatoes scores. It’s likely that we are looking at missing values, so we’ll need to look through our data. It’s worth noting here that since Rotten Tomatoes rates movies with a percentage, R has identified this as a factor variable, instead of a numeric one. To get proper results, we needed to convert these values to numbers with `as.numeric()`.
+
+![](https://i.imgur.com/pUMPbdq.png)
+
+Just as suspected, many of the movies are missing Rotten Tomatoes scores. There are a number of ways we could handle this, one of which is to create a new dataframe of movies that are missing this rating, and to then remove these movies from our original dataframe.
+
```

Comment:
maybe sneak in a link that covers general data imputation. the reader may wonder where to go from here... in some case, replace by the mean/mode, predict the NA etc